src/com/duckduckgo/mobile/android/adapters/OnboardingAdapter.java

```diff
@@ -22,41 +17,26 @@
 
 public class OnboardingAdapter extends FragmentPagerAdapter {
 
-    private static final int NUM_PAGES = 5;
-    //private boolean useMiniLayout = false;
-    public OnboardingAdapter(FragmentManager fm) {
-        super(fm);
-        //NUM_PAGES = 5;
-        //Log.e("NUM_PAGES", "numPages: "+NUM_PAGES);
-    }
+    private final BaseOnboardingFragment[] items;
 
-    private OnboardingAdapter(FragmentManager fm, boolean useMiniLayout) {
+    public OnboardingAdapter(FragmentManager fm) {
         super(fm);
-        //this.useMiniLayout = useMiniLayout;
-        //NUM_PAGES = useMiniLayout ? 4 : 5;
-        //Log.e("NUM_PAGES", "numPages: "+NUM_PAGES);
+        items = new BaseOnboardingFragment[] {
```

Comment:
Storing pojo/model data in an array in your adapter is a good idea, however the fragments themselves are heavy and should usually be constructed on demand. I wonder if this would work better if:
• the OnboardingPageConfiguration data was stored in an array in the adapter rather than the fragments themselves
• there was a single OnboardingPageFragment class (rather than the BaseOnboaringFramgnet + subclasses) which had the OnboardingPageConfiguration injected in via fragment args

I think this is a quite win as the BaseOnboaringFragment subclasses look identical other than the data.